diff --git a/CelestiaAppComponent/CelestiaAppComponent.vcxproj b/CelestiaAppComponent/CelestiaAppComponent.vcxproj
index 36af562..c70b35f 100644
--- a/CelestiaAppComponent/CelestiaAppComponent.vcxproj
+++ b/CelestiaAppComponent/CelestiaAppComponent.vcxproj
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.props" Condition="Exists('..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.props')" />
+  <Import Project="..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.props" Condition="Exists('..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.props')" />
   <PropertyGroup Label="Globals">
     <CppWinRTOptimized>true</CppWinRTOptimized>
     <CppWinRTRootNamespaceAutoMerge>true</CppWinRTRootNamespaceAutoMerge>
@@ -280,6 +281,7 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
     <Import Project="..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.targets" Condition="Exists('..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.targets')" />
+    <Import Project="..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.targets" Condition="Exists('..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.targets')" />
   </ImportGroup>
   <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
     <PropertyGroup>
@@ -287,5 +289,7 @@
     </PropertyGroup>
     <Error Condition="!Exists('..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.props'))" />
     <Error Condition="!Exists('..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.targets'))" />
+    <Error Condition="!Exists('..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.props'))" />
+    <Error Condition="!Exists('..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.targets'))" />
   </Target>
 </Project>
\ No newline at end of file
diff --git a/CelestiaAppComponent/packages.config b/CelestiaAppComponent/packages.config
index 1b8cfbd..97815be 100644
--- a/CelestiaAppComponent/packages.config
+++ b/CelestiaAppComponent/packages.config
@@ -1,4 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="Microsoft.Windows.CppWinRT" version="2.0.240405.15" targetFramework="native" />
+  <package id="celestia-uwp-dependencies" version="1.0.10" targetFramework="native" />
 </packages>
\ No newline at end of file
diff --git a/CelestiaComponent/CelestiaComponent.vcxproj b/CelestiaComponent/CelestiaComponent.vcxproj
index 19d050b..259c5cc 100644
--- a/CelestiaComponent/CelestiaComponent.vcxproj
+++ b/CelestiaComponent/CelestiaComponent.vcxproj
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project DefaultTargets="Build" ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.props" Condition="Exists('..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.props')" />
+  <Import Project="..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.props" Condition="Exists('..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.props')" />
   <ItemGroup Label="ProjectConfigurations">
     <ProjectConfiguration Include="Debug|ARM">
       <Configuration>Debug</Configuration>
@@ -1155,6 +1156,7 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
     <Import Project="..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.targets" Condition="Exists('..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.targets')" />
+    <Import Project="..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.targets" Condition="Exists('..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.targets')" />
   </ImportGroup>
   <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
     <PropertyGroup>
@@ -1162,5 +1164,7 @@
     </PropertyGroup>
     <Error Condition="!Exists('..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.props'))" />
     <Error Condition="!Exists('..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.targets'))" />
+    <Error Condition="!Exists('..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.props'))" />
+    <Error Condition="!Exists('..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-uwp-dependencies.1.0.10\build\native\celestia-uwp-dependencies.targets'))" />
   </Target>
 </Project>
\ No newline at end of file
diff --git a/CelestiaComponent/packages.config b/CelestiaComponent/packages.config
index 020eb5a..b181d08 100644
--- a/CelestiaComponent/packages.config
+++ b/CelestiaComponent/packages.config
@@ -1,4 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="Microsoft.Windows.CppWinRT" version="2.0.240405.15" targetFramework="native" />
+  <package id="celestia-uwp-dependencies" version="1.0.10" targetFramework="native" />
 </packages>
\ No newline at end of file
