diff --git a/CelestiaAppComponent/CelestiaAppComponent.vcxproj b/CelestiaAppComponent/CelestiaAppComponent.vcxproj
index 83c43da..b5580ef 100644
--- a/CelestiaAppComponent/CelestiaAppComponent.vcxproj
+++ b/CelestiaAppComponent/CelestiaAppComponent.vcxproj
@@ -7,6 +7,7 @@
   <Import Project="..\packages\Microsoft.Windows.SDK.BuildTools.MSIX.1.7.20250829.1\build\Microsoft.Windows.SDK.BuildTools.MSIX.props" Condition="Exists('..\packages\Microsoft.Windows.SDK.BuildTools.MSIX.1.7.20250829.1\build\Microsoft.Windows.SDK.BuildTools.MSIX.props')" />
   <Import Project="..\packages\Microsoft.Windows.CppWinRT.2.0.250303.1\build\native\Microsoft.Windows.CppWinRT.props" Condition="Exists('..\packages\Microsoft.Windows.CppWinRT.2.0.250303.1\build\native\Microsoft.Windows.CppWinRT.props')" />
   <Import Project="..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.6901\build\Microsoft.Windows.SDK.BuildTools.props" Condition="Exists('..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.6901\build\Microsoft.Windows.SDK.BuildTools.props')" />
+  <Import Project="..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.props" Condition="Exists('..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.props')" />
   <PropertyGroup Label="Globals">
     <CppWinRTOptimized>true</CppWinRTOptimized>
     <CppWinRTRootNamespaceAutoMerge>true</CppWinRTRootNamespaceAutoMerge>
@@ -324,6 +325,7 @@
     <Import Project="..\packages\Microsoft.WindowsAppSDK.InteractiveExperiences.1.8.250906004\build\native\Microsoft.WindowsAppSDK.InteractiveExperiences.targets" Condition="Exists('..\packages\Microsoft.WindowsAppSDK.InteractiveExperiences.1.8.250906004\build\native\Microsoft.WindowsAppSDK.InteractiveExperiences.targets')" />
     <Import Project="..\packages\Microsoft.WindowsAppSDK.Foundation.1.8.250906002\build\native\Microsoft.WindowsAppSDK.Foundation.targets" Condition="Exists('..\packages\Microsoft.WindowsAppSDK.Foundation.1.8.250906002\build\native\Microsoft.WindowsAppSDK.Foundation.targets')" />
     <Import Project="..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.targets" Condition="Exists('..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.targets')" />
+    <Import Project="..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.targets" Condition="Exists('..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.targets')" />
   </ImportGroup>
   <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
     <PropertyGroup>
@@ -344,5 +346,7 @@
     <Error Condition="!Exists('..\packages\Microsoft.WindowsAppSDK.Foundation.1.8.250906002\build\native\Microsoft.WindowsAppSDK.Foundation.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.WindowsAppSDK.Foundation.1.8.250906002\build\native\Microsoft.WindowsAppSDK.Foundation.targets'))" />
     <Error Condition="!Exists('..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.props'))" />
     <Error Condition="!Exists('..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.targets'))" />
+    <Error Condition="!Exists('..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.props'))" />
+    <Error Condition="!Exists('..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.targets'))" />
   </Target>
 </Project>
\ No newline at end of file
diff --git a/CelestiaAppComponent/packages.config b/CelestiaAppComponent/packages.config
index a1902d8..af5920f 100644
--- a/CelestiaAppComponent/packages.config
+++ b/CelestiaAppComponent/packages.config
@@ -8,4 +8,5 @@
   <package id="Microsoft.WindowsAppSDK.Foundation" version="1.8.250906002" targetFramework="native" />
   <package id="Microsoft.WindowsAppSDK.InteractiveExperiences" version="1.8.250906004" targetFramework="native" />
   <package id="Microsoft.WindowsAppSDK.WinUI" version="1.8.250906003" targetFramework="native" />
+  <package id="celestia-windows-arm64-dependencies" version="1.0.9" targetFramework="native" />
 </packages>
\ No newline at end of file
diff --git a/CelestiaComponent/CelestiaComponent.vcxproj b/CelestiaComponent/CelestiaComponent.vcxproj
index 02ba5cf..aa46bf9 100644
--- a/CelestiaComponent/CelestiaComponent.vcxproj
+++ b/CelestiaComponent/CelestiaComponent.vcxproj
@@ -7,6 +7,7 @@
   <Import Project="..\packages\Microsoft.Windows.SDK.BuildTools.MSIX.1.7.20250829.1\build\Microsoft.Windows.SDK.BuildTools.MSIX.props" Condition="Exists('..\packages\Microsoft.Windows.SDK.BuildTools.MSIX.1.7.20250829.1\build\Microsoft.Windows.SDK.BuildTools.MSIX.props')" />
   <Import Project="..\packages\Microsoft.Windows.CppWinRT.2.0.250303.1\build\native\Microsoft.Windows.CppWinRT.props" Condition="Exists('..\packages\Microsoft.Windows.CppWinRT.2.0.250303.1\build\native\Microsoft.Windows.CppWinRT.props')" />
   <Import Project="..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.6901\build\Microsoft.Windows.SDK.BuildTools.props" Condition="Exists('..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.6901\build\Microsoft.Windows.SDK.BuildTools.props')" />
+  <Import Project="..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.props" Condition="Exists('..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.props')" />
   <ItemGroup Label="ProjectConfigurations">
     <ProjectConfiguration Include="Debug|ARM">
       <Configuration>Debug</Configuration>
@@ -1204,6 +1205,7 @@
     <Import Project="..\packages\Microsoft.WindowsAppSDK.InteractiveExperiences.1.8.250906004\build\native\Microsoft.WindowsAppSDK.InteractiveExperiences.targets" Condition="Exists('..\packages\Microsoft.WindowsAppSDK.InteractiveExperiences.1.8.250906004\build\native\Microsoft.WindowsAppSDK.InteractiveExperiences.targets')" />
     <Import Project="..\packages\Microsoft.WindowsAppSDK.Foundation.1.8.250906002\build\native\Microsoft.WindowsAppSDK.Foundation.targets" Condition="Exists('..\packages\Microsoft.WindowsAppSDK.Foundation.1.8.250906002\build\native\Microsoft.WindowsAppSDK.Foundation.targets')" />
     <Import Project="..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.targets" Condition="Exists('..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.targets')" />
+    <Import Project="..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.targets" Condition="Exists('..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.targets')" />
   </ImportGroup>
   <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
     <PropertyGroup>
@@ -1225,5 +1227,7 @@
     <Error Condition="!Exists('..\packages\Microsoft.WindowsAppSDK.Foundation.1.8.250906002\build\native\Microsoft.WindowsAppSDK.Foundation.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.WindowsAppSDK.Foundation.1.8.250906002\build\native\Microsoft.WindowsAppSDK.Foundation.targets'))" />
     <Error Condition="!Exists('..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.props'))" />
     <Error Condition="!Exists('..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.targets'))" />
+    <Error Condition="!Exists('..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.props'))" />
+    <Error Condition="!Exists('..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.targets'))" />
   </Target>
 </Project>
\ No newline at end of file
diff --git a/CelestiaComponent/packages.config b/CelestiaComponent/packages.config
index d53fe46..90e4821 100644
--- a/CelestiaComponent/packages.config
+++ b/CelestiaComponent/packages.config
@@ -9,4 +9,5 @@
   <package id="Microsoft.WindowsAppSDK.Foundation" version="1.8.250906002" targetFramework="native" />
   <package id="Microsoft.WindowsAppSDK.InteractiveExperiences" version="1.8.250906004" targetFramework="native" />
   <package id="Microsoft.WindowsAppSDK.WinUI" version="1.8.250906003" targetFramework="native" />
+  <package id="celestia-windows-arm64-dependencies" version="1.0.9" targetFramework="native" />
 </packages>
\ No newline at end of file
diff --git a/CelestiaWinUI/CelestiaWinUI.vcxproj b/CelestiaWinUI/CelestiaWinUI.vcxproj
index c1dfa1a..bc40a60 100644
--- a/CelestiaWinUI/CelestiaWinUI.vcxproj
+++ b/CelestiaWinUI/CelestiaWinUI.vcxproj
@@ -7,6 +7,7 @@
   <Import Project="..\packages\Microsoft.Windows.SDK.BuildTools.MSIX.1.7.20250829.1\build\Microsoft.Windows.SDK.BuildTools.MSIX.props" Condition="Exists('..\packages\Microsoft.Windows.SDK.BuildTools.MSIX.1.7.20250829.1\build\Microsoft.Windows.SDK.BuildTools.MSIX.props')" />
   <Import Project="..\packages\Microsoft.Windows.CppWinRT.2.0.250303.1\build\native\Microsoft.Windows.CppWinRT.props" Condition="Exists('..\packages\Microsoft.Windows.CppWinRT.2.0.250303.1\build\native\Microsoft.Windows.CppWinRT.props')" />
   <Import Project="..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.6901\build\Microsoft.Windows.SDK.BuildTools.props" Condition="Exists('..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.6901\build\Microsoft.Windows.SDK.BuildTools.props')" />
+  <Import Project="..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.props" Condition="Exists('..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.props')" />
   <PropertyGroup Label="Globals">
     <CppWinRTOptimized>true</CppWinRTOptimized>
     <CppWinRTRootNamespaceAutoMerge>true</CppWinRTRootNamespaceAutoMerge>
@@ -745,6 +746,7 @@
     <Import Project="..\packages\Microsoft.WindowsAppSDK.InteractiveExperiences.1.8.250906004\build\native\Microsoft.WindowsAppSDK.InteractiveExperiences.targets" Condition="Exists('..\packages\Microsoft.WindowsAppSDK.InteractiveExperiences.1.8.250906004\build\native\Microsoft.WindowsAppSDK.InteractiveExperiences.targets')" />
     <Import Project="..\packages\Microsoft.WindowsAppSDK.Foundation.1.8.250906002\build\native\Microsoft.WindowsAppSDK.Foundation.targets" Condition="Exists('..\packages\Microsoft.WindowsAppSDK.Foundation.1.8.250906002\build\native\Microsoft.WindowsAppSDK.Foundation.targets')" />
     <Import Project="..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.targets" Condition="Exists('..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.targets')" />
+    <Import Project="..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.targets" Condition="Exists('..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.targets')" />
   </ImportGroup>
   <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
     <PropertyGroup>
@@ -766,5 +768,7 @@
     <Error Condition="!Exists('..\packages\Microsoft.WindowsAppSDK.Foundation.1.8.250906002\build\native\Microsoft.WindowsAppSDK.Foundation.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.WindowsAppSDK.Foundation.1.8.250906002\build\native\Microsoft.WindowsAppSDK.Foundation.targets'))" />
     <Error Condition="!Exists('..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.props'))" />
     <Error Condition="!Exists('..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.WindowsAppSDK.WinUI.1.8.250906003\build\native\Microsoft.WindowsAppSDK.WinUI.targets'))" />
+    <Error Condition="!Exists('..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.props'))" />
+    <Error Condition="!Exists('..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-windows-arm64-dependencies.1.0.9\build\native\celestia-windows-arm64-dependencies.targets'))" />
   </Target>
 </Project>
\ No newline at end of file
diff --git a/CelestiaWinUI/packages.config b/CelestiaWinUI/packages.config
index fd42a0d..6794f1a 100644
--- a/CelestiaWinUI/packages.config
+++ b/CelestiaWinUI/packages.config
@@ -9,4 +9,5 @@
   <package id="Microsoft.WindowsAppSDK.Foundation" version="1.8.250906002" targetFramework="native" />
   <package id="Microsoft.WindowsAppSDK.InteractiveExperiences" version="1.8.250906004" targetFramework="native" />
   <package id="Microsoft.WindowsAppSDK.WinUI" version="1.8.250906003" targetFramework="native" />
+  <package id="celestia-windows-arm64-dependencies" version="1.0.9" targetFramework="native" />
 </packages>
\ No newline at end of file
