diff --git a/CelestiaAppComponent/CelestiaAppComponent.vcxproj b/CelestiaAppComponent/CelestiaAppComponent.vcxproj
index 4c24b70..29346bc 100644
--- a/CelestiaAppComponent/CelestiaAppComponent.vcxproj
+++ b/CelestiaAppComponent/CelestiaAppComponent.vcxproj
@@ -3,6 +3,7 @@
   <Import Project="..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.props" Condition="Exists('..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.props')" />
   <Import Project="..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.props" Condition="Exists('..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.props')" />
   <Import Project="..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.1742\build\Microsoft.Windows.SDK.BuildTools.props" Condition="Exists('..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.1742\build\Microsoft.Windows.SDK.BuildTools.props')" />
+  <Import Project="..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.props" Condition="Exists('..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.props')" />
   <PropertyGroup Label="Globals">
     <CppWinRTOptimized>true</CppWinRTOptimized>
     <CppWinRTRootNamespaceAutoMerge>true</CppWinRTRootNamespaceAutoMerge>
@@ -86,28 +87,28 @@
   <PropertyGroup Label="UserMacros" />
   <PropertyGroup />
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
-    <VcpkgTriplet>x64-windows</VcpkgTriplet>
+    <VcpkgTriplet>x64-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
-    <VcpkgTriplet>x64-windows</VcpkgTriplet>
+    <VcpkgTriplet>x64-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
-    <VcpkgTriplet>x86-windows</VcpkgTriplet>
+    <VcpkgTriplet>x86-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
-    <VcpkgTriplet>x86-windows</VcpkgTriplet>
+    <VcpkgTriplet>x86-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Debug|ARM64'">
-    <VcpkgTriplet>arm64-windows</VcpkgTriplet>
+    <VcpkgTriplet>arm64-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Release|ARM64'">
-    <VcpkgTriplet>arm64-windows</VcpkgTriplet>
+    <VcpkgTriplet>arm64-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Debug|ARM'">
-    <VcpkgTriplet>arm-windows</VcpkgTriplet>
+    <VcpkgTriplet>arm-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Release|ARM'">
-    <VcpkgTriplet>arm-windows</VcpkgTriplet>
+    <VcpkgTriplet>arm-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <ItemDefinitionGroup>
     <ClCompile>
@@ -292,6 +293,7 @@
     <Import Project="..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.1742\build\Microsoft.Windows.SDK.BuildTools.targets" Condition="Exists('..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.1742\build\Microsoft.Windows.SDK.BuildTools.targets')" />
     <Import Project="..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.targets" Condition="Exists('..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.targets')" />
     <Import Project="..\packages\Microsoft.Web.WebView2.1.0.3124.44\build\native\Microsoft.Web.WebView2.targets" Condition="Exists('..\packages\Microsoft.Web.WebView2.1.0.3124.44\build\native\Microsoft.Web.WebView2.targets')" />
+    <Import Project="..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.targets" Condition="Exists('..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.targets')" />
   </ImportGroup>
   <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
     <PropertyGroup>
@@ -304,5 +306,7 @@
     <Error Condition="!Exists('..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.props'))" />
     <Error Condition="!Exists('..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.targets'))" />
     <Error Condition="!Exists('..\packages\Microsoft.Web.WebView2.1.0.3124.44\build\native\Microsoft.Web.WebView2.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.Web.WebView2.1.0.3124.44\build\native\Microsoft.Web.WebView2.targets'))" />
+    <Error Condition="!Exists('..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.props'))" />
+    <Error Condition="!Exists('..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.targets'))" />
   </Target>
 </Project>
\ No newline at end of file
diff --git a/CelestiaAppComponent/packages.config b/CelestiaAppComponent/packages.config
index 6485506..e226b6d 100644
--- a/CelestiaAppComponent/packages.config
+++ b/CelestiaAppComponent/packages.config
@@ -4,4 +4,5 @@
   <package id="Microsoft.Windows.CppWinRT" version="2.0.240405.15" targetFramework="native" />
   <package id="Microsoft.Windows.SDK.BuildTools" version="10.0.26100.1742" targetFramework="native" />
   <package id="Microsoft.WindowsAppSDK" version="1.6.250228001" targetFramework="native" />
+  <package id="celestia-windows-release-dependencies" version="1.0.5" targetFramework="native" />
 </packages>
\ No newline at end of file
diff --git a/CelestiaComponent/CelestiaComponent.vcxproj b/CelestiaComponent/CelestiaComponent.vcxproj
index 2bd732d..ac52abe 100644
--- a/CelestiaComponent/CelestiaComponent.vcxproj
+++ b/CelestiaComponent/CelestiaComponent.vcxproj
@@ -3,6 +3,7 @@
   <Import Project="..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.props" Condition="Exists('..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.props')" />
   <Import Project="..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.props" Condition="Exists('..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.props')" />
   <Import Project="..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.1742\build\Microsoft.Windows.SDK.BuildTools.props" Condition="Exists('..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.1742\build\Microsoft.Windows.SDK.BuildTools.props')" />
+  <Import Project="..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.props" Condition="Exists('..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.props')" />
   <ItemGroup Label="ProjectConfigurations">
     <ProjectConfiguration Include="Debug|ARM">
       <Configuration>Debug</Configuration>
@@ -158,28 +159,28 @@
     <IncludePath>$(ProjectDir);$(ProjectDir)..\..\Celestia\src;$(ProjectDir)$(GeneratedFilesDir);$(IncludePath)</IncludePath>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Debug|ARM64'">
-    <VcpkgTriplet>arm64-windows</VcpkgTriplet>
+    <VcpkgTriplet>arm64-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Release|ARM64'">
-    <VcpkgTriplet>arm64-windows</VcpkgTriplet>
+    <VcpkgTriplet>arm64-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Debug|ARM'">
-    <VcpkgTriplet>arm-windows</VcpkgTriplet>
+    <VcpkgTriplet>arm-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Release|ARM'">
-    <VcpkgTriplet>arm-windows</VcpkgTriplet>
+    <VcpkgTriplet>arm-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
-    <VcpkgTriplet>x86-windows</VcpkgTriplet>
+    <VcpkgTriplet>x86-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
-    <VcpkgTriplet>x86-windows</VcpkgTriplet>
+    <VcpkgTriplet>x86-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
-    <VcpkgTriplet>x64-windows</VcpkgTriplet>
+    <VcpkgTriplet>x64-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
-    <VcpkgTriplet>x64-windows</VcpkgTriplet>
+    <VcpkgTriplet>x64-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
     <ClCompile>
@@ -1184,6 +1185,7 @@
     <Import Project="..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.1742\build\Microsoft.Windows.SDK.BuildTools.targets" Condition="Exists('..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.1742\build\Microsoft.Windows.SDK.BuildTools.targets')" />
     <Import Project="..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.targets" Condition="Exists('..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.targets')" />
     <Import Project="..\packages\Microsoft.Web.WebView2.1.0.3124.44\build\native\Microsoft.Web.WebView2.targets" Condition="Exists('..\packages\Microsoft.Web.WebView2.1.0.3124.44\build\native\Microsoft.Web.WebView2.targets')" />
+    <Import Project="..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.targets" Condition="Exists('..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.targets')" />
   </ImportGroup>
   <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
     <PropertyGroup>
@@ -1196,5 +1198,7 @@
     <Error Condition="!Exists('..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.props'))" />
     <Error Condition="!Exists('..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.targets'))" />
     <Error Condition="!Exists('..\packages\Microsoft.Web.WebView2.1.0.3124.44\build\native\Microsoft.Web.WebView2.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.Web.WebView2.1.0.3124.44\build\native\Microsoft.Web.WebView2.targets'))" />
+    <Error Condition="!Exists('..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.props'))" />
+    <Error Condition="!Exists('..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.targets'))" />
   </Target>
 </Project>
\ No newline at end of file
diff --git a/CelestiaComponent/packages.config b/CelestiaComponent/packages.config
index a0a271e..3f855fc 100644
--- a/CelestiaComponent/packages.config
+++ b/CelestiaComponent/packages.config
@@ -4,4 +4,5 @@
   <package id="Microsoft.Windows.CppWinRT" version="2.0.240405.15" targetFramework="native" />
   <package id="Microsoft.Windows.SDK.BuildTools" version="10.0.26100.1742" targetFramework="native" />
   <package id="Microsoft.WindowsAppSDK" version="1.6.250228001" targetFramework="native" />
+  <package id="celestia-windows-release-dependencies" version="1.0.5" targetFramework="native" />
 </packages>
\ No newline at end of file
diff --git a/CelestiaWinUI/CelestiaWinUI.vcxproj b/CelestiaWinUI/CelestiaWinUI.vcxproj
index 6685240..22dc7ae 100644
--- a/CelestiaWinUI/CelestiaWinUI.vcxproj
+++ b/CelestiaWinUI/CelestiaWinUI.vcxproj
@@ -3,6 +3,7 @@
   <Import Project="..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.props" Condition="Exists('..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.props')" />
   <Import Project="..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.props" Condition="Exists('..\packages\Microsoft.Windows.CppWinRT.2.0.240405.15\build\native\Microsoft.Windows.CppWinRT.props')" />
   <Import Project="..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.1742\build\Microsoft.Windows.SDK.BuildTools.props" Condition="Exists('..\packages\Microsoft.Windows.SDK.BuildTools.10.0.26100.1742\build\Microsoft.Windows.SDK.BuildTools.props')" />
+  <Import Project="..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.props" Condition="Exists('..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.props')" />
   <PropertyGroup Label="Globals">
     <CppWinRTOptimized>true</CppWinRTOptimized>
     <CppWinRTRootNamespaceAutoMerge>true</CppWinRTRootNamespaceAutoMerge>
@@ -85,28 +86,28 @@
     </ClCompile>
   </ItemDefinitionGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Debug|ARM64'">
-    <VcpkgTriplet>arm64-windows</VcpkgTriplet>
+    <VcpkgTriplet>arm64-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Release|ARM64'">
-    <VcpkgTriplet>arm64-windows</VcpkgTriplet>
+    <VcpkgTriplet>arm64-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Debug|ARM'">
-    <VcpkgTriplet>arm-windows</VcpkgTriplet>
+    <VcpkgTriplet>arm-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Release|ARM'">
-    <VcpkgTriplet>arm-windows</VcpkgTriplet>
+    <VcpkgTriplet>arm-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
-    <VcpkgTriplet>x86-windows</VcpkgTriplet>
+    <VcpkgTriplet>x86-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
-    <VcpkgTriplet>x86-windows</VcpkgTriplet>
+    <VcpkgTriplet>x86-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
-    <VcpkgTriplet>x64-windows</VcpkgTriplet>
+    <VcpkgTriplet>x64-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <PropertyGroup Label="Vcpkg" Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
-    <VcpkgTriplet>x64-windows</VcpkgTriplet>
+    <VcpkgTriplet>x64-windows-release</VcpkgTriplet>
   </PropertyGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)'=='Debug'">
     <ClCompile>
@@ -679,6 +680,7 @@
     <Import Project="..\packages\Microsoft.Windows.ImplementationLibrary.1.0.240803.1\build\native\Microsoft.Windows.ImplementationLibrary.targets" Condition="Exists('..\packages\Microsoft.Windows.ImplementationLibrary.1.0.240803.1\build\native\Microsoft.Windows.ImplementationLibrary.targets')" />
     <Import Project="..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.targets" Condition="Exists('..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.targets')" />
     <Import Project="..\packages\Microsoft.Web.WebView2.1.0.3124.44\build\native\Microsoft.Web.WebView2.targets" Condition="Exists('..\packages\Microsoft.Web.WebView2.1.0.3124.44\build\native\Microsoft.Web.WebView2.targets')" />
+    <Import Project="..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.targets" Condition="Exists('..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.targets')" />
   </ImportGroup>
   <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
     <PropertyGroup>
@@ -692,5 +694,7 @@
     <Error Condition="!Exists('..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.props'))" />
     <Error Condition="!Exists('..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.WindowsAppSDK.1.6.250228001\build\native\Microsoft.WindowsAppSDK.targets'))" />
     <Error Condition="!Exists('..\packages\Microsoft.Web.WebView2.1.0.3124.44\build\native\Microsoft.Web.WebView2.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.Web.WebView2.1.0.3124.44\build\native\Microsoft.Web.WebView2.targets'))" />
+    <Error Condition="!Exists('..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.props'))" />
+    <Error Condition="!Exists('..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\celestia-windows-release-dependencies.1.0.5\build\native\celestia-windows-release-dependencies.targets'))" />
   </Target>
 </Project>
\ No newline at end of file
diff --git a/CelestiaWinUI/packages.config b/CelestiaWinUI/packages.config
index 02e21b6..75e67ad 100644
--- a/CelestiaWinUI/packages.config
+++ b/CelestiaWinUI/packages.config
@@ -5,4 +5,5 @@
   <package id="Microsoft.Windows.ImplementationLibrary" version="1.0.240803.1" targetFramework="native" />
   <package id="Microsoft.Windows.SDK.BuildTools" version="10.0.26100.1742" targetFramework="native" />
   <package id="Microsoft.WindowsAppSDK" version="1.6.250228001" targetFramework="native" />
+  <package id="celestia-windows-release-dependencies" version="1.0.5" targetFramework="native" />
 </packages>
\ No newline at end of file
